python/ray/autoscaler/command_runner.py

```diff
@@ -279,19 +293,17 @@ def run_rsync_down(self, source, target, options=None):
                 target,
             ])
         except Exception as e:
-            logger.warning(self.log_prefix +
-                           "rsync failed: '{}'. Falling back to 'kubectl cp'"
-                           .format(e))
-            self.run_cp_down(source, target)
-
-    def run_cp_down(self, source, target):
-        if target.startswith("~"):
-            target = "/root" + target[1:]
-
-        self.process_runner.check_call(self.kubectl + [
-            "cp", "{}/{}:{}".format(self.namespace, self.node_id, source),
-            target
-        ])
+            warnings.warn(
+                self.log_prefix +
+                "rsync failed: '{}'. Falling back to 'kubectl cp'".format(e),
+                UserWarning)
+            if target.startswith("~"):
+                target = "/root" + target[1:]
+
+            self.process_runner.check_call(self.kubectl + [
```

Comment:
Is it safe to use `warnings` instead of `logger.warning`?

Reply:
@richardliaw

Reply:
yes